src/reader.cpp

```diff
@@ -1136,17 +1136,11 @@ static history_pager_result_t history_pager_search(const HistorySharedPtr &histo
     size_t page_size = std::max(termsize_last().height / 2 - 2, (rust::isize)12);
 
     rust::Box<completion_list_t> completions = new_completion_list();
+
     rust::Box<HistorySearch> search =
-        rust_history_search_new(history, search_string.c_str(), history_search_type_t::Contains,
+        rust_history_search_new(history, search_string.c_str(), history_search_type_t::ContainsGlob,
                                 smartcase_flags(search_string), history_index);
     bool next_match_found = search->go_to_next_match(direction);
-    if (!next_match_found) {
-        // If there were no matches, try again with subsequence search
-        search = rust_history_search_new(history, search_string.c_str(),
-                                         history_search_type_t::ContainsSubsequence,
-                                         smartcase_flags(search_string), history_index);
-        next_match_found = search->go_to_next_match(direction);
-    }
```

Comment:
removing the subsequence match fallback is somewhat unrelated.
I don't think we should do it in this PR, if at all.

The reason why people may expect subsequence matching is because the completion pager has it.
This overloaded behavior is weird but it usually doesn't matter.

We should make this fallback search for `parse_util_unescape_wildcards(search_string)` but without
the ANY_CHAR and ANY_STRING because they are implied by subsequence.

Reply:
I agree. The existing behaviour is restored in https://github.com/fish-shell/fish-shell/pull/10143/commits/f73c4a36dc130355e107204226f135725969c7c3.

Note that I think it makes little sense to fallback to subsequence search if the user include wildcard in their search term, so I added a check for that.